src/block_cache/block_mgr.c

```diff
@@ -60,7 +140,23 @@ static int
 __bm_checkpoint(
   WT_BM *bm, WT_SESSION_IMPL *session, WT_ITEM *buf, WT_CKPT *ckptbase, bool data_checksum)
 {
-    return (__wt_block_checkpoint(session, bm->block, buf, ckptbase, data_checksum));
+    WT_BLOCK *block;
+
+    block = bm->block;
+
+    WT_RET(__wt_block_checkpoint(session, block, buf, ckptbase, data_checksum));
+
+    /*
+     * After a checkpoint, we can close previously writeable tiered objects. The block's reference
+     * count was set when the block was originally opened, switching didn't change it and this is
+     * the close.
+     */
+    if (block->ckpt_drain != NULL) {
```

Comment:
This conditional is not necessary as the first line of `bm_drain_and_close` will check that. The single function call overhead is not something to worry about on this path.

Reply:
Fixed in 1a0093c59.